app/services/fetch_resource_service.rb

```diff
@@ -3,7 +3,7 @@
 class FetchResourceService < BaseService
   include JsonLdHelper
 
-  ACCEPT_HEADER = 'application/activity+json, application/ld+json; profile="https://www.w3.org/ns/activitystreams", text/html;q=0.1'
+  ACCEPT_HEADER = 'application/ld+json; profile="https://www.w3.org/ns/activitystreams", application/activity+json, text/html;q=0.1'
```

Comment:
this change doesn't make any sense to me, the two headers should be semantically equivalent. I don't think we should make this change, it just increases fragility in the ecosystem (encouraging servers to do byte for byte comparisons instead of parsing headers properly

Reply:
> the two headers should be semantically equivalent

They are not; see the spec excerpt i quoted in the OP.

Spec REQUIRES request be made with a JSON-LD header with the activitystream profile and merely suggests to perhaps also accept the `activity+json` one when receiving requests

This also doesn’t encourage byte-for-byte comparisons, since while the JSON-LD type with the ActivityStreams 2.0 profile is required, it is allowed to set further additional profiles, other attributes and ofc there may be several content types like here.  
Logically it makes the most sense to me though to put the canonical and only required header at the start of the list, which implies it to be the preferred type

Reply:
> > the two headers should be semantically equivalent
> 
> They are not (always); see the spec excerpt i quoted in the OP and #22720

Care to elaborate? The excerpt in the OP doesn't seem to me to prohibit additional media type as long as the header value contains `application/ld+json; profile="https://www.w3.org/ns/activitystreams"`, and it doesn't seem to care about the ordering of the values either (well, of course it doesn't, since the HTTP spec doesn't assign any meaning to lexical ordering of the `Accept` header values).

Reply:
I think i might have misunderstood which "header" nightpool was referring to. I assumed the headers in question were `application/ld+json; profile="..."` and `application/activity+json`.

But reading your further reply and rereading the original, perhaps the compound header before and after the patch was meant? If so, yes they are semantically equivalent. Putting the spec-compliant mandated header first is a cosmetic hcnage if you will, for better consistency across the code and to communicate its importance to human readers.